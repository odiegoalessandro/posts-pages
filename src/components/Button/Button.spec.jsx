import { cleanup, render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { Button } from ".";

afterEach(cleanup);

describe("<Button />", () => {
  it('should render the button with the text "test"', () => {
    const fn = jest.fn();
    
    render(<Button text="test" onClick={fn} />);

    const button = screen.getByRole('button', { name: /test/i });
    
    userEvent.click(button);
    
    expect(fn).toHaveBeenCalledTimes(1);
    expect(button).toBeInTheDocument();
  });


  it('should be disabled when disabled prop is true', () => {
    render(<Button disabled={true} text="test" />);

    const button = screen.getByRole('button', { name: /test/i });
    expect(button).toBeDisabled();
  });

  it('should be match snapshot', () => {
    const { asFragment } = render(<Button text="test" />);

    expect(asFragment()).toMatchSnapshot();
  });
});