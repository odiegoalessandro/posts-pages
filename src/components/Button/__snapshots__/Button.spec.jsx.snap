// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should be match snapshot 1`] = `
<DocumentFragment>
  <button
    class="button"
  >
    test
  </button>
</DocumentFragment>
`;
